herbert_core/utilities/misc/hlp_deserialise.m

```diff
@@ -140,7 +140,9 @@
 
     [i, pos] = read_bytes(m, pos, 'uint64', nElem);
     [j, pos] = read_bytes(m, pos, 'uint64', nElem);
-
+    % +1 is to align with C API which indexes from 0, not 1
+    i = double(i + 1)
```

Comment:
why are these integers being converted to double?

Reply:
Because from the CI:
```Name:     test_cpp_serialise::test_ser_complex_sparse_array

/home/jenkins/workspace/PACE-neutrons/Herbert/PR-Scientific-Linux-7-2018b/herbert_core/utilities/misc/hlp_deserialise.m at <a href="matlab: opentoline('/home/jenkins/workspace/PACE-neutrons/Herbert/PR-Scientific-Linux-7-2018b/herbert_core/utilities/misc/hlp_deserialise.m',156)">line 156</a>
/home/jenkins/workspace/PACE-neutrons/Herbert/PR-Scientific-Linux-7-2018b/herbert_core/utilities/misc/hlp_deserialise.m at <a href="matlab: opentoline('/home/jenkins/workspace/PACE-neutrons/Herbert/PR-Scientific-Linux-7-2018b/herbert_core/utilities/misc/hlp_deserialise.m',51)">line 51</a>
/home/jenkins/workspace/PACE-neutrons/Herbert/PR-Scientific-Linux-7-2018b/herbert_core/utilities/misc/hlp_deserialise.m at <a href="matlab: opentoline('/home/jenkins/workspace/PACE-neutrons/Herbert/PR-Scientific-Linux-7-2018b/herbert_core/utilities/misc/hlp_deserialise.m',30)">line 30</a>
/home/jenkins/workspace/PACE-neutrons/Herbert/PR-Scientific-Linux-7-2018b/_test/test_utilities/test_cpp_serialise.m at <a href="matlab: opentoline('/home/jenkins/workspace/PACE-neutrons/Herbert/PR-Scientific-Linux-7-2018b/_test/test_utilities/test_cpp_serialise.m',502)">line 502</a>

Undefined function 'sparse' for input arguments of type 'uint64'.```

Reply:
That's unexpected. The MATLAB documentation defines `i` and `j` arguments as being: 

```Data Types: single | double | int8 | int16 | int32 | int64 | uint8 | uint16 | uint32 | uint64 | logical```

https://www.mathworks.com/help/matlab/ref/sparse.html

Reply:
Unexpected to me too, but I'm seeing if it fixes it, then we can work out what to do.